cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -285,6 +286,14 @@ func StartControllers(s *options.CMServer, kubeClient *client.Client, kubeconfig
 		glog.Infof("Starting %s apis", groupVersion)
 		if containsResource(resources, "horizontalpodautoscalers") {
 			glog.Infof("Starting horizontal pod controller.")
+			// Make a copy of conf for scale client so that we don't accidentally set group version for other clients.
+			scaleConf := *(restclient.AddUserAgent(kubeconfig, "horizontal-pod-autoscaler"))
+			gv, err := unversioned.ParseGroupVersion(groupVersion)
+			if err != nil {
+				glog.Fatalf("Failed to parse group version %s: %v", groupVersion, err)
+			}
+			scaleConf.GroupVersion = &gv
+			scaleClient := scaleclient.NewForConfigOrDie(&scaleConf)
```

Comment:
scaleClient is a dynamic client and hpaClient is a generated client that points to core and extensions API (unfortunately internal client right now). We need both.